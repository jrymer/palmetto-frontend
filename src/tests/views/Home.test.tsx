import React from 'react';

import HomeView from '../../views/home';
import { createSnapshot, fireEvent, render, screen } from '../testWrapper';

describe('HomeView tests', () => {
  it('HomeView snapshot is valid with base props', () => {
    const snapshot = createSnapshot(<HomeView />, true);
    expect(snapshot).toMatchSnapshot();
  });
  it('Can type', () => {
    const searchValue = 'hello world';
    render(<HomeView />, true);
    const input = screen.getByRole('textbox');
    fireEvent.change(input, { target: { value: searchValue } });
    expect(input).toHaveValue(searchValue);
  });
  it('Has geolocation button', () => {
    render(<HomeView />, true);
    const button = screen.getByText(/use current location\?/i);
    expect(button).toBeInTheDocument();
  });
});
