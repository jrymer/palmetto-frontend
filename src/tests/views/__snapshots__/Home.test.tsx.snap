// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView tests HomeView snapshot is valid with base props 1`] = `
<div
  className="sc-fEOsli etEBoq"
>
  <div
    className="sc-ksZaOG kdKPBT"
  >
    <div
      className="sc-hAZoDl nRnXU"
    >
      <div
        className="sc-gKXOVf bYsPSI"
      >
        <div
          className="sc-iBkjds ffxseL"
        >
          <input
            className="sc-ftvSup cSQcVc"
            onChange={[Function]}
            placeholder="Search for a cities weather by typing the cities name."
            value="Boulder, CO, USA"
          />
          <img
            className="sc-kDDrLX iXqnFh arrow-up"
            onClick={[Function]}
            src="down.png"
          />
          <img
            className="sc-kDDrLX iXqnFh"
            onClick={[Function]}
            src="cancel.png"
          />
        </div>
      </div>
      <div
        className="sc-evZas bHogbH"
      >
        <div
          className="sc-breuTD bKyzzD"
          onClick={[Function]}
        >
          Imperial
        </div>
        <div
          className="sc-breuTD fbmTqK"
          onClick={[Function]}
        >
          Metric
        </div>
      </div>
      <div
        className="sc-iqcoie gRqzmc"
        disabled={false}
        onClick={[Function]}
        role="button"
        width="var(--col-width)"
      >
        <span
          className="sc-crXcEl cXKdGn"
        >
          Use current location?
        </span>
      </div>
    </div>
    <div
      className="sc-fnykZs knWTCK"
    >
      <div
        className="sc-iqcoie oABfL"
        disabled={false}
        onClick={[Function]}
        role="button"
        width="100%"
      >
        <span
          className="sc-crXcEl cXKdGn"
        >
          Search
        </span>
      </div>
    </div>
  </div>
  <div
    className="sc-idiyUo gmimwu"
  >
    <div
      className="sc-bczRLJ kRwXGP"
    />
  </div>
</div>
`;
